app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,47 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = @user.posts.includes(:comments)
+    @posts = @user.posts
+    @current_user = current_user
   end
 
   def show
     @post = Post.find(params[:id])
-    @user = @post.author
-    @comments = @post.comments
+    @current_user = current_user
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @like = Like.new(author_id: params[:user_id], post_id: @post.id)
+
+    if @like.save
+      redirect_to user_post_path(user_id: params[:user_id], id: params[:id]), notice: 'Like added successfully'
+    else
+      flash.now[:alert] = 'Failed to add like'
+      render 'posts/show'
+    end
+  end
```

Comment:
- [ ] I think it will be better if you extract this from your Post controller and create a Likes controller where you will then add it. This way, your code will be modular and readable.